File: libraries/leo2-common/src/main/java/org/deku/leo2/rest/services/v1/RoutingService.java
package org.deku.leo2.rest.services.v1;

import com.wordnik.swagger.annotations.Api;
import com.wordnik.swagger.annotations.ApiOperation;
import com.wordnik.swagger.annotations.ApiParam;
import org.deku.leo2.rest.entities.v1.Routing;
import org.deku.leo2.rest.entities.v1.RoutingRequest;

import javax.ws.rs.Consumes;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.core.MediaType;

//import org.deku.leo2.rest.entities.v1.RoutingVia;

/**
 * Created by masc on 17.09.14.
 */
@Path("v1/routing")
@Consumes(MediaType.APPLICATION_JSON)
@Api(value = "Routing operations")
public interface RoutingService {
    @POST
    @Path("/request")
    @ApiOperation(value = "Determ routing and via")
    Routing request(@ApiParam(value = "Routing request") RoutingRequest routingRequest);
}

File: leoz-common/src/main/java/org/deku/leoz/service/v1/RoutingService.kt
package org.deku.leoz.service.v1

import io.swagger.annotations.*
import org.deku.leoz.service.entity.v1.Routing
import org.deku.leoz.service.entity.v1.RoutingRequest
import org.deku.leoz.service.entity.v1.ServiceError

import javax.ws.rs.Consumes
import javax.ws.rs.POST
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType

//import org.deku.leoz.rest.entities.v1.RoutingVia;

/**
 * Created by masc on 17.09.14.
 */
@javax.ws.rs.Path("v1/routing")
@javax.ws.rs.Produces(javax.ws.rs.core.MediaType.APPLICATION_JSON)
@javax.ws.rs.Consumes(javax.ws.rs.core.MediaType.APPLICATION_JSON)
@io.swagger.annotations.Api(value = "Routing operations")
interface RoutingService {
    /**
     * Routing service specific error codes
     */
    enum class ErrorCode private constructor(private val mValue: Int) {
        ROUTE_NOT_AVAILABLE_FOR_GIVEN_PARAMETER(1000)
    }

    @javax.ws.rs.POST
    @javax.ws.rs.Path("/request")
    @io.swagger.annotations.ApiOperation(value = "Request routing information")
    @io.swagger.annotations.ApiResponses(*arrayOf(
            io.swagger.annotations.ApiResponse(code = 400, message = "Bad request/parameter", response = ServiceError::class))
    )
    fun request(@io.swagger.annotations.ApiParam(value = "Routing request") routingRequest: org.deku.leoz.service.entity.v1.RoutingRequest): org.deku.leoz.service.entity.v1.Routing
}
